app/post.go

```diff
@@ -674,6 +675,126 @@ func (a *App) GetPostsAroundPost(postId, channelId string, offset, limit int, be
 	return a.Srv.Store.Post().GetPostsAfter(channelId, postId, limit, offset)
 }
 
+func (a *App) GetPostAfterTime(channelId string, time int64) (*model.Post, *model.AppError) {
+	return a.Srv.Store.Post().GetPostAfterTime(channelId, time)
+}
+
+func (a *App) GetPostBeforeTime(channelId string, time int64) (*model.Post, *model.AppError) {
+	return a.Srv.Store.Post().GetPostBeforeTime(channelId, time)
+}
+
+func (a *App) GetNextPostIdFromPostList(postList *model.PostList) string {
+	if len(postList.Order) > 0 {
+		firstPostId := postList.Order[0]
+		firstPost := postList.Posts[firstPostId]
+		nextPost, err := a.GetPostAfterTime(firstPost.ChannelId, firstPost.CreateAt)
```

Comment:
Given we only use the `Id` here, I wonder if there's a potential performance improvement to expose a `GetPostIdAfterTime`. I believe it could be satisfied completely from the `idx_posts_channel_id_delete_at_create_at` index, skipping the page hit for the unused post data.

Reply:
There's instance that the [Post object is needed](https://github.com/mattermost/mattermost-server/pull/11486/files#diff-317b12c12f71e311f13813107f4d8550R773) and so I maintain `GetPostAfterTime` but for those using `Id` only, I exposed `GetPostIdAfterTime` and `GetPostIdBeforeTime`.